Modules/CustomCommand.cs

```diff
@@ -0,0 +1,131 @@
+    /// </summary>
+    public class CustomCommand
+    {
+        /// <summary>
+        /// A result than can be returned from a role operation.
+        /// </summary>
+        protected class RoleOperationErrorResult
+        {
+            /// <summary>
+            /// The name of the role that caused the error.
+            /// </summary>
+            public string RoleName { get; set; }
+            /// <summary>
+            /// Does the user already have the role?
+            /// </summary>
+            public bool HasRole { get; set; }
+        }
+
+        /// <summary>
+        /// The message that will be printed by the command.
+        /// </summary>
+        [JsonProperty("message")]
+        public string Message { get; set; } = string.Empty;
+        /// <summary>
+        /// Should the sent message be sent as an embed?
+        /// </summary>
+        [JsonProperty("embed")]
+        public bool Embed { get; set; } = false;
+        /// <summary>
+        /// The role operations that will be perfomed, seperated by ','.
+        /// </summary>
+        [JsonProperty("roleOperations")]
+        public string RoleOperations { get; set; } = null;
+
+        /// <summary>
+        /// Executes the command with the specified user in the specified channel.
+        /// </summary>
+        public async Task Execute(IGuildUser user, IMessageChannel channel)
+        {
+            RoleOperationErrorResult[] results = await PerformRoleOperations(user);
+            if (results.Length > 0)
+            {
+                // If there are any error results, print the errors.
+                EmbedBuilder eb = new EmbedBuilder()
+                    .WithTitle("Info")
+                    .WithDescription(string.Join('\n', results.Select(r => r.HasRole 
+                        ? $"You already have the role {r.RoleName}!"
+                        : $"You don't have the role {r.RoleName}!")));
+
+                await channel.SendMessageAsync(string.Empty, false, eb);
+            }
+            else
+            {
+                // If not, print the regular command message.
+                await PerformMessageOperation(channel);
+            }
+        }
+
+        /// <summary>
+        /// Prints the message for this command to the specified channel.
+        /// </summary>
+        private async Task PerformMessageOperation(IMessageChannel channel)
+        {
+            if (string.IsNullOrEmpty(Message)) return;
+
+            // Conditionally encapsulate the message in an embed
+            if (Embed) await channel.SendMessageAsync(string.Empty, false, new EmbedBuilder().WithDescription(Message));
+            else await channel.SendMessageAsync(Message);
+        }
+        /// <summary>
+        /// Adds roles prefixed with '+' to the user and removes roles prefixed with '-' from the user.
+        /// </summary>
+        private async Task<RoleOperationErrorResult[]> PerformRoleOperations(IGuildUser user)
```

Comment:
It would be nicer if the return would be something like `RoleOperationResult` which would have field for `IsError` and `ErrorMessage`. This way it looks like this method will always produce an error of some kind.